mpmath/functions/zeta.py

```diff
@@ -553,17 +550,13 @@ def zeta(ctx, s, a=1, derivative=0, method=None, **kwargs):
         #        pass
         if abs(im) > 500*prec and 10*re < prec and derivative <= 4 or \
             method == 'riemann-siegel':
-            try:   #  py2.4 compatible try block
-                try:
-                    if verbose:
-                        print("zeta: Attempting to use the Riemann-Siegel algorithm")
-                    return ctx.rs_zeta(s, derivative, **kwargs)
-                except NotImplementedError:
-                    if verbose:
-                        print("zeta: Could not use the Riemann-Siegel algorithm")
-                    pass
-            finally:
-                ctx.prec = prec
+            try:
+                if verbose:
+                    print("zeta: Attempting to use the Riemann-Siegel algorithm")
+                return ctx.rs_zeta(s, derivative, **kwargs)
+            except NotImplementedError:
+                if verbose:
+                    print("zeta: Could not use the Riemann-Siegel algorithm")
```

Comment:
maybe its explained in the commit messages, but why is the `finally: ctx.prec = prec` dropped here?

I don't immediately see it being changed, but I also haven't studied this code yet...

Reply:
I dug a little deeper: `rs_zeta` calls `rs_offline` which certainly changes the `.prec`.

Reply:
It seems to be changed in 6437c031ab43d3d6c15a682198ab641fcbc0a643, but the commit log doesn't offer an explanation

Reply:
Indeed, probably that was blindly copied from #597 and is my fault.  The workaround was introduced in b3bc191.